### PR TITLE
chore(ci): bump lotus version in devnet check

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

I have recently seen quite a few ci failures in the devnet checks job, this PR tries to upgrade lotus to the latest and see if the issue get mitigated (even not we need the upgrade anyway)

Changes introduced in this pull request:

- bump lotus version in devnet check CI job

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: scripts/devnet/lotus.dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Lotus binaries image, to be used in the local devnet with Forest.
-FROM golang:1.20.7-bullseye AS lotus-builder
+FROM golang:1.20-bullseye AS lotus-builder
```
Comment (hanabi1224) [inline scripts/devnet/lotus.dockerfile:2]: I believe we should use the latest patch version of Go1.20

**Comment by LesnyRumcajs** [comment]
Could you please elaborate on the errors that you've encountered?

**Comment by hanabi1224** [comment]
@LesnyRumcajs a few links below

https://github.com/ChainSafe/forest/actions/runs/7527552013/job/20488198997
https://github.com/ChainSafe/forest/actions/runs/7382999656/job/20083619234

**Review by lemmih — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
> @LesnyRumcajs a few links below
> 
> https://github.com/ChainSafe/forest/actions/runs/7527552013/job/20488198997 https://github.com/ChainSafe/forest/actions/runs/7382999656/job/20083619234

Thanks. We are missing logs, I'll add them here https://github.com/ChainSafe/forest/pull/3890

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**